languages/scala/reference/README.md

```diff
@@ -0,0 +1,193 @@
+# Scala reference
+
+## Concepts
+
+The Scala concept exercises are based on concepts. The list below contains the concepts that have been identified for the Scala language.
+
+### Language-unique
+
+- Pattern matching
+- Functional Programming
+- Immutability
+
+### Functional
+
+- [Functions][functions]
+  - [Pure functions][pure_functions]
+  - [Higher-order functions][higher_order_functions]
+  - [Anonymous functions][anonymous_functions]
+  - [Nested functions][nested_functions]
+  - [Function composition][function_composition]
+  - [Partial application][partial_application]
+  - [Recursion][recursion]
+    - Tail recursion
+  - Currying
+  - Implicit returns
+  - Type annotations
+- [Immutability][immutability]
+- [Pattern matching][pattern_matching]
+  - Discards
+  - Guard clauses
+- [Type inference][type_inference]
+  - Automatic generalisation
+- [Pipelines][pipelines]
+- [Expression-oriented (not statement oriented)][expression_oriented]
+
+### Object-oriented
+
+The core OO features an Scala developer should know about are:
+
+- [Classes][classes]
+- [case class][case class]
+- [Encapsulation][encapsulation]
+- [Objects][objects]
+- [Interfaces][interfaces]
+  - Object expressions
+- [Methods][methods]
+  - Constructors
+  - Apply
+  - Unapply
+  - Method arguments
+    - Named arguments
+    - [Optional arguments][optional_arguments]
+    - Out parameters
+- Fields
+- Properties
+- Type extensions
```

Comment:
Is this the same as traits?

Reply:
I would guess you meant `enrich my library` (aka. `pimp my library`) pattern or `mixins`, which are both good things to mention. Although `enrich my library` goes hand in hand with `implicit class` below